server/raft_test.go

```diff
@@ -936,10 +938,23 @@ func TestNRGNoResetOnAppendEntryResponse(t *testing.T) {
 	})
 	require_NoError(t, err)
 
-	// Generate an append entry that the subscriber above can respond to.
 	c.waitOnAllCurrent()
+
+	// Temporarily prevent followers to respond
+	locked := rg.lockFollowers()
+
+	// Generate an append entry that the subscriber above can respond to.
 	lsm.proposeDelta(5)
-	rg.waitOnTotal(t, 5)
+
+	// Expect the leader to step down
+	for isLeader == true {
```

Comment:
Might be better to use `checkFor()` here with a timeout set, otherwise this could potentially stick for the entire timeout of the test if something goes wrong.

Reply:
I'm not sure I understand this comment. require_ChanRead() already takes a timeout.